### PR TITLE
CAM: replace CAM test detection by globbing with a cmake-generated file

### PR DESCRIPTION
**Pull request by davidgilkaufman**
@Connor9220 here's a fix to the problem where my new globbing strategy runs stale tests that are left behind in the build directory

Instead of globbing, I have cmake generate a file that lists the test modules enumerated in the Tests_SRCS variable. When you switch branches, this list changes, and the file is regenerated with the new list. Any stale test files left behind are ignored.

---

Separately, while testing my PR, I noticed a bug where TestCAMApp runs fewer tests than CAMTests, even though they should be the same (TestCAMApp just imports everything from CAMTests). I tracked it down to the `module` variable in my loop in `__init__.py`; this variable gets picked up by the test runner and run as extra tests, duplicates of the tests from the last iterated module.

(fwiw I think this bug was present when I first introduced enumeration-based test running, but perhaps it was inconsequential then because the enumeration order happened to include a last module that didn't include any tests? I do think I validated my original PR by checking that those two commands ran the same tests, so perhaps/probably this bug only became a problem with this PR. I haven't checked if it duplicates a few tests on `main`)

Anyways, I've fixed this issue by using a local function in __init__.py to ensure that local variables do not pollute the global namespace. With this fix, `-t CAMTests` does not run duplicate tests, and `-t CAMTests` and `-t TestCAMApp` run the same number of tests.

### CODE REVIEW

**Comment by Connor9220** [comment]
I had discovered another way too..  simply delete the files from the directory before the copy..

```
diff --git a/src/Mod/CAM/CMakeLists.txt b/src/Mod/CAM/CMakeLists.txt
index 58876e1cd8b..46b90a386de 100644
--- a/src/Mod/CAM/CMakeLists.txt
+++ b/src/Mod/CAM/CMakeLists.txt
@@ -769,6 +769,8 @@ ADD_CUSTOM_TARGET(Tests ALL
 )
 
 fc_copy_sources(PathScripts "${CMAKE_BINARY_DIR}/Mod/CAM" ${all_files})
+# Drop the previously-copied tests so renamed/deleted ones do not linger in the build tree
+file(REMOVE_RECURSE "${CMAKE_BINARY_DIR}/Mod/CAM/CAMTests")
 fc_copy_sources(Tests "${CMAKE_BINARY_DIR}/Mod/CAM" ${test_files})
 
 # Copy machine templates to build directory
```
